src/ai/backend/manager/registry.py

```diff
@@ -1123,6 +1119,14 @@ async def enqueue_session(
                 if tpu is not None:
                     raise InvalidAPIParameters("Client upgrade required to use TPUs (v19.03+).")
 
+            # Check if the user has allocated an "imbalanced" shared memory amount.
+            if shmem >= requested_slots["mem"]:
```

Comment:
Let's make the ratio configurable, via etcd's scoped configurations (e.g., possible to override by resource groups).